gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AgentDirectoryMapper.java

```diff
@@ -221,12 +223,13 @@ private String getOrganizationName(String name) {
     }
 
     private CodeableConcept getText(CV code) {
-        var codeableConcept = new CodeableConcept();
+
         if (code != null) {
             if (StringUtils.isNotEmpty(code.getOriginalText())) {
-                return codeableConcept.setText(code.getOriginalText());
+                return createCodeableConcept(code.getCode(), code.getCodeSystem(),
+                                             code.getDisplayName(), code.getOriginalText());
             } else if (StringUtils.isNotEmpty(code.getDisplayName())) {
-                return codeableConcept.setText(code.getDisplayName());
+                return createCodeableConcept(code.getCode(), code.getCodeSystem(), code.getDisplayName());
```

Comment:
I think I mentioned my dislike of this API previously, but it makes the mistake made here where the code and code system are put in as the wrong arguments harder to spot/notice.

If the API remained as:

`code.setCode(code.getCodeSystem()).setCodeSystem(code.getCode())`

It would be a lot clearer that a mistake had been made.